worker/src/fetcher.rs

```diff
@@ -1,33 +1,69 @@
+use crate::{env::EnvBinding, RequestInit, Result};
+#[cfg(feature = "http")]
+use std::convert::TryInto;
 use wasm_bindgen::{JsCast, JsValue};
 use wasm_bindgen_futures::JsFuture;
 
-use crate::{env::EnvBinding, Request, RequestInit, Response, Result};
-
+#[cfg(feature = "http")]
+use crate::{HttpRequest, HttpResponse};
+use crate::{Request, Response};
 /// A struct for invoking fetch events to other Workers.
 pub struct Fetcher(worker_sys::Fetcher);
 
+#[cfg(not(feature = "http"))]
```

Comment:
Ah, good catch, did not realize there are two implementations.

Reply:
I started going down this route and it turned out to be a can of worms, so I think this will need to be done in a follow up PR. 

Mainly, it would involve a lot of changes in the many places that the global fetcher is used.

It also forced me to think about being able to convert between old and new types by reference. Supporting that is pretty painful and I'm on the fence whether it is even reasonable. So we may want to change `Fetch::fetch_with_request` to take an owned request if that seems ok. The only use case I can think of having a reference here is making multiple fetches with the same request, which must also not have a body. I'm not sure how common that is.